lib/HeadersGeneration/ModelToHeaderPipe.cpp

```diff
@@ -76,3 +77,40 @@ static pipeline::RegisterDefaultConstructibleContainer<ModelHeaderFileContainer>
 } // namespace revng::pipes
 
 static pipeline::RegisterPipe<revng::pipes::ModelToHeader> Y;
+
+namespace revng::pypeline::pipes {
+
+ObjectDependencies ModelToHeader::run(const Model &TheModel,
+                                      const Request &Incoming,
+                                      const Request &Outgoing,
+                                      llvm::StringRef DynamicConfig,
+                                      RootBuffer &Buffer) {
+  if (Outgoing[0].empty())
+    return {};
+
+  revng_assert(Outgoing[0].size() == 1);
+  ObjectID Target = *Outgoing[0][0];
+  const model::Binary &Binary = *TheModel.get().get();
+  revng::Tracking::clearAndResume(Binary);
+  std::unique_ptr<llvm::raw_ostream> Out = Buffer.getOStream(Target);
+  ptml::ModelCBuilder
+    B(*Out,
+      Binary,
+      /* EnableTaglessMode = */ false,
+      { .EnableStackFrameInlining = revng::options::EnableStackFrameInlining,
+        .EnablePrintingOfTheMaximumEnumValue = true,
+        .ExplicitTargetPointerSize = getExplicitPointerSize(Binary) });
+  ptml::HeaderBuilder(B).printModelHeader();
+  Out->flush();
+
+  ObjectDependencies Result(1);
```

Comment:
`1`?

Reply:
`ModelToHeaderPipe` uses one container, so the `ObjectDependencies` (which is just `std::vector<std::vector<std::pair<ObjectID, ModelPath>>>`) needs to be created with one element.